api/metadata/serializers.py

```diff
@@ -0,0 +1,133 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Q
+from rest_framework import serializers
+
+from .models import (
+    FIELD_VALUE_MAX_LENGTH,
+    FieldType,
+    Metadata,
+    MetadataField,
+    MetadataModelField,
+)
+
+
+class MetadataFieldSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = MetadataField
+        fields = ("id", "name", "type", "description", "organisation")
+
+
+class MetaDataModelFieldSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = MetadataModelField
+        fields = ("id", "field", "is_required")
+
+
+class FieldDataField(serializers.Field):
+    """
+    Custom field to handle the different types of data supported by the metadata field
+    """
+
+    def to_internal_value(self, data):
+        data_type = type(data).__name__
+
+        if data_type not in [field.value for field in FieldType.__members__.values()]:
+            raise serializers.ValidationError("Invalid data type")
+
+        if data_type == str and len(data) > FIELD_VALUE_MAX_LENGTH:
+            raise serializers.ValidationError(
+                f"Value string is too long. Must be less than {FIELD_VALUE_MAX_LENGTH} character"
+            )
+        return data
+
+    def to_representation(self, value):
+        return value
+
+
+class MetadataListSerializer(serializers.ListSerializer):
+    def save(self, **kwargs):
```

Comment:
I'm a little confused what we're doing here - is this some kind of bulk create functionality?